common/djangoapps/student/views.py

```diff
@@ -1230,30 +1232,45 @@ def login_user(request, error=""):  # pylint: disable=too-many-statements,unused
 
         # if we didn't find this username earlier, the account for this email
         # doesn't exist, and doesn't have a corresponding password
+        msg = None
         if username != "":
             if settings.FEATURES['SQUELCH_PII_IN_LOGS']:
                 loggable_id = user_found_by_email_lookup.id if user_found_by_email_lookup else "<unknown>"
                 AUDIT_LOG.warning(u"Login failed - password for user.id: {0} is invalid".format(loggable_id))
             else:
                 AUDIT_LOG.warning(u"Login failed - password for {0} is invalid".format(email))
 
-        # Labster changes
-        # If there was login error, suggest student to login to region server
-        regions = configuration_helpers.get_value('REGIONS', settings.REGIONS)
-        # List of country codes: https://dev.maxmind.com/geoip/legacy/codes/iso3166/
-        current_region = request.session.get('country_code')
-        if current_region in regions.keys():
-            region_info = regions[current_region]
-            msg = _(
-                'If you are a student in {region_name}, your university may now be using the {region_code} version of '
-                'Labster. Please try to log into <a href="{region_login_url}">{region_name}</a> region.'
-            ).format(
-                region_name=region_info['name'],
-                region_code=region_info['region_code'],
-                region_login_url=region_info['login_url'],
-            )
+        # Start: Added by Labster
         else:
-            msg = _('Email or password is incorrect.')
+            # If there was login error, check whether user with the given email exists in one of the regions. If yes,
+            # redirect to the appropriate login url or suggest student to login to the appropriate region server.
+            base_url = request.build_absolute_uri('/')[:-1]
+            regions = configuration_helpers.get_value('REGIONS', settings.REGIONS)
+            for region_code, region in regions.items():
+                login_url = region['login_url']
+                # Take out the url of this current region in the list of regions
+                if base_url in login_url:
+                    del regions[region_code]
```

Comment:
@lukyfebrianto why do you change a dict you're working with? It may introduce unpredictable consequences in the future, can we simply use `continue` instead.
```py
for x in y:
  if x in z:
    continue
  if some_func(x):
    do_semthing()
 ...
```
or
```py
for x in y:
  if x not in z and some_func(x):
     do_semthing()
 ...
```

Reply:
yes @polesye I need to delete those region because to avoid warning message displayed in the current region server.
updated using continue
fyi @polesye

Reply:
I'm still quite concerned changing mutable settings objects.
I'd suggest to use list comprehensions instead: 
```py
regions = {region_code: region for region_code, region in regions.items() if base_url not in region['login_url']}
for region in regions.values():
  if is_user_exist(region, email):
    ...
```

Reply:
ahhh, i see your point now.
updated